TheHiveV5/tests/test_add_observable.py

```diff
@@ -0,0 +1,441 @@
+        "extraData": {},
+        "ignoreSimilarity": False,
+    },
+    {
+        "_id": "~40968200",
+        "_type": "Observable",
+        "_createdBy": "testapi@thehive.local",
+        "_createdAt": 1760086562768,
+        "dataType": "domain",
+        "data": "www.binance.com",
+        "startDate": 1760086562768,
+        "tlp": 2,
+        "tlpLabel": "AMBER",
+        "pap": 2,
+        "papLabel": "AMBER",
+        "tags": [],
+        "ioc": False,
+        "sighted": False,
+        "reports": {},
+        "extraData": {},
+        "ignoreSimilarity": False,
+    },
+    {
+        "_id": "~81924184",
+        "_type": "Observable",
+        "_createdBy": "testapi@thehive.local",
+        "_createdAt": 1760086564549,
+        "dataType": "url",
+        "data": "http://malicious.example1/path",
+        "startDate": 1760086564549,
+        "tlp": 2,
+        "tlpLabel": "AMBER",
+        "pap": 2,
+        "papLabel": "AMBER",
+        "tags": [],
+        "ioc": False,
+        "sighted": False,
+        "reports": {},
+        "extraData": {},
+        "ignoreSimilarity": False,
+    },
+]
+
+
+def test_add_observables_action_success():
+    action = TheHiveCreateObservableV5()
+    action.module.configuration = {
+        "base_url": "https://thehive-project.org",
+        "apikey": "LOREM",
+        "organisation": "SEKOIA",
+    }
+
+    with requests_mock.Mocker() as mock_requests:
+        # mock_requests.post(url="https://thehive-project.org/api/v1/alert/{ALERT_ID}/observable", status_code=200, json=HIVE_OUTPUT)
+        url = f"https://thehive-project.org/api/v1/alert/{ALERT_ID}/observable"
+        mock_requests.post(url=url, status_code=200, json=HIVE_OUTPUT)
+
+        result: List[OutputObservable] = action.run({"alert_id": ALERT_ID, "events": EVENTS})
+        assert result is not None
+        # Assert that there is a result with dataType == "domain" and data == "www.binance.com"
+        # OutputObservable is likely a Dict[str, Any], so .get is valid
+        # Flatten result if it is a list of lists
+        flat_result = [item for sublist in result for item in (sublist if isinstance(sublist, list) else [sublist])]
+        assert any(obs["dataType"] == "domain" and obs["data"] == "www.binance.com" for obs in flat_result)
+        assert any(obs["dataType"] == "ip" and obs["data"] == "192.168.0.1" for obs in flat_result)
+
+
+def test_add_observables_action_api_error(requests_mock):
+    # mock_alert = requests_mock.post(url="https://thehive-project.org/api/v1/alert/{ALERT_ID}/observable", status_code=500)
+    url = f"https://thehive-project.org/api/v1/alert/{ALERT_ID}/observable"
+    mock_alert = requests_mock.post(url=url, status_code=500)
+
+    action = TheHiveCreateObservableV5()
+    action.module.configuration = {
+        "base_url": "https://thehive-project.org",
+        "apikey": "LOREM",
+        "organisation": "SEKOIA",
+    }
+
+    result = action.run({"alert_id": ALERT_ID, "events": EVENTS})
```

Comment:
The test fails here as a `thehive4py.errors.TheHiveError` is raised and stop the test


```suggestion
    with pytest.raises(thehive4py.errors.TheHiveError)
        result = action.run({"alert_id": ALERT_ID, "events": EVENTS})
```
With the import of `thehive4py`